scene/resources/visual_shader.cpp

```diff
@@ -4217,11 +4217,10 @@ void VisualShaderNodeGroupBase::set_inputs(const String &p_inputs) {
 
 		int port_idx = arr[0].to_int();
 		int port_type = arr[1].to_int();
-		String port_name = arr[2];
 
 		Port port;
 		port.type = (PortType)port_type;
-		port.name = port_name;
+		port.name = arr[2];
 		input_ports[port_idx] = port;
```

Comment:
These two do not look like an improvement to me. Sure, we don't need to cast anything here for the string, but the way the code is laid out makes it very easy to see the composition of the parsed input string before it is assigned to the corresponding fields and properties. I would keep it as is.